crossbar/worker/test/test_router.py

```diff
@@ -257,3 +267,194 @@ def test_start_router_component_wrong_baseclass(self):
+        r.start_router_transport(
+            "component1",
+            {
+                u"type": u"web",
+                u"endpoint": {
+                    u"type": u"tcp",
+                    u"port": 8080
+                },
+                u"paths": {
+                    u"/": {
+                        "module": u"crossbar.worker.test.test_router",
+                        "object": u"hello",
+                        "type": u"wsgi"
+                    },
+                    u"json": {
+                        "type": u"json",
+                        "value": {}
+                    }
+                }
+            })
+
+        # Make a request to the /json endpoint, which is technically a child of
+        # the WSGI app, but is not served by WSGI.
+        d = treq.get("http://localhost:8080/json", reactor=temp_reactor)
+        d.addCallback(treq.content)
+        d.addCallback(self.assertEqual, b"{}")
+        d.addCallback(lambda _: temp_reactor.stop())
+
+        temp_reactor.run()
+
+        return d
+
+    def test_threads(self):
+        """
+        A basic WSGI app can be ran, with subresources
+        """
+        temp_reactor = SelectReactor()
+        r = router.RouterWorkerSession(config=self.config,
+                                       reactor=temp_reactor)
+
+        # Open the transport
+        transport = FakeWAMPTransport(r)
+        r.onOpen(transport)
+
+        realm_config = {
+            u"name": u"realm1",
+            u'roles': []
+        }
+
+        threads = 20
+
+        r.start_router_realm("realm1", realm_config)
+        r.start_router_transport(
+            "component1",
+            {
+                u"type": u"web",
+                u"endpoint": {
+                    u"type": u"tcp",
+                    u"port": 8080
+                },
+                u"paths": {
+                    u"/": {
+                        "module": u"crossbar.worker.test.test_router",
+                        "object": u"sleep",
+                        "type": u"wsgi",
+                        "maxthreads": threads,
+                    }
+                }
+            })
+
+        deferreds = []
+        results = []
+
+        for i in range(threads):
+
+            d = treq.get("http://localhost:8080/", reactor=temp_reactor)
+            d.addCallback(treq.content)
+            d.addCallback(results.append)
+            deferreds.append(d)
+
+        d = defer.DeferredList(deferreds)
+        d.addCallback(lambda _: self.assertIn(str(threads).encode('ascii'),
+                                              results))
+        d.addCallback(lambda _: temp_reactor.stop())
+
+        temp_reactor.run()
+
+        return d
+
+
+def hello(environ, start_response):
+    """
+    A super dumb WSGI app for testing.
+    """
+    start_response('200 OK', [('Content-Type', 'text/html')])
+    return [b'hello!']
+
+
+# Ugh global state, but it's just for a test...
+count = []
+
```

Comment:
WSGI apps are a callable that takes an environment and a `start_response` callable.

The other thing to note is that 'object' is actually JSON -- I've just done it as a raw dictionary for simplicity ;)